x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_events_query.ts

```diff
@@ -151,3 +166,66 @@ export const buildEventsSearchQuery = ({
   }
   return searchQuery;
 };
+
+export const buildEqlSearchRequest = (
+  query: string,
+  index: string[],
+  from: string,
+  to: string,
+  size: number,
+  timestampOverride: TimestampOverrideOrUndefined,
+  exceptionLists: ExceptionListItemSchema[],
+  eventCategoryOverride: string | undefined
+): estypes.EqlSearchRequest => {
+  const defaultTimeFields = ['@timestamp'];
+  const timestamps =
+    timestampOverride != null ? [timestampOverride, ...defaultTimeFields] : defaultTimeFields;
+  const docFields = timestamps.map((tstamp) => ({
+    field: tstamp,
+    format: 'strict_date_optional_time',
+  }));
+
+  // Assume that `indices.query.bool.max_clause_count` is at least 1024 (the default value),
+  // allowing us to make 1024-item chunks of exception list items.
+  // Discussion at https://issues.apache.org/jira/browse/LUCENE-4835 indicates that 1024 is a
+  // very conservative value.
+  const exceptionFilter = buildExceptionFilter({
+    lists: exceptionLists,
+    excludeExceptions: true,
+    chunkSize: 1024,
+  });
+
+  const rangeFilter = buildTimeRangeFilter({ to, from, timestampOverride });
```

Comment:
This is the core difference in this PR - `buildTimeRangeFilter` extracts the logic to build a time range filter that falls back to `@timestamp` if the timestamp override doesn't exist in a document. Previously the time range filter for EQL rules (https://github.com/elastic/kibana/pull/127989/files#diff-1284eea43a7a651ad24f50fa2a3b02761b5c6ea7bde83c7ebaca5b47eaa79383L96-L103) did not fall back.